### PR TITLE
chore: convert more files to TypeScript

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Convert the following files to TypeScript:
- `lib/common/buffer-utils.js`
- `lib/common/clipboard-utils.js`
- `lib/common/error-utils.js`
- `lib/common/web-view-methods.js`
- `lib/browser/devtools.js`

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

File: lib/common/buffer-utils.ts [hunk 1]
```diff
@@ -19,16 +14,16 @@ const typedArrays = {
   Float64Array
 }
 
-function getType (value) {
+function getType (value: ArrayBuffer | ArrayBufferView) {
```
Comment (nornagon) [inline lib/common/buffer-utils.ts]:
```suggestion
function getType (value: BufferSource) {
```

https://heycam.github.io/webidl/#BufferSource

File: lib/common/buffer-utils.ts [hunk 2]
```diff
@@ -38,19 +33,25 @@ function getBuffer (value) {
...
+  data: Buffer;
+  length: number;
+}
+
+export function bufferToMeta (value: ArrayBuffer | ArrayBufferView): BufferMeta {
   return {
     type: getType(value),
     data: getBuffer(value),
-    length: value.length
+    length: (value as Buffer).length
```
Comment (nornagon) [inline lib/common/buffer-utils.ts]: this doesn't seem right. `ArrayBuffer` doesn't have a `length` property (it has `byteLength`).

File: lib/common/error-utils.ts [hunk 3]
```diff
@@ -10,10 +8,15 @@ const constructors = new Map([
   [URIError.name, URIError]
 ])
 
-exports.deserialize = function (error) {
+interface ElectronError extends Error {
+  from?: string;
+  cause?: ElectronError;
+}
+
+export function deserialize (error: Electron.SerializedError) {
```
Comment (nornagon) [inline lib/common/error-utils.ts]: Add a return type?

File: lib/common/error-utils.ts [hunk 4]
```diff
@@ -22,7 +25,7 @@ exports.deserialize = function (error) {
   return error
 }
 
-exports.serialize = function (error) {
+export function serialize (error: ElectronError) {
```
Comment (nornagon) [inline lib/common/error-utils.ts]: Return type here too please

File: lib/common/buffer-utils.ts (hunk 2 above)
Comment (miniak) [reply lib/common/buffer-utils.ts]: we only need `length` when we are serializing `Int8Array`, etc.
when deserializing `Buffer` and `ArrayBuffer`, `length` is not used

File: lib/common/buffer-utils.ts [hunk 5]
```diff
@@ -19,16 +14,16 @@ const typedArrays = {
   Float64Array
 }
 
-function getType (value) {
+function getType (value: BufferSource) {
```
Comment (nornagon) [inline lib/common/buffer-utils.ts]: hm... actually this should also include `Buffer` which `BufferSource` doesn't.

maybe `type BufferLike = ArrayBuffer | ArrayBufferView | Buffer`?

File: lib/common/buffer-utils.ts (hunk 2 above)
Comment (nornagon) [reply lib/common/buffer-utils.ts]: Then `BufferMeta` should have `length: number | undefined`?

File: lib/common/buffer-utils.ts (hunk 5 above)
Comment (miniak) [reply lib/common/buffer-utils.ts]: `Buffer` is now actually `Uint8Array`, but it's treated specially to keep it transferred as `Buffer`

**Comment by miniak** [comment]
@nornagon can you please check again?

File: lib/common/buffer-utils.ts (hunk 2 above)
Comment (nornagon) [reply lib/common/buffer-utils.ts]: Please also add a comment here, this is confusing & looks like a bug without the appropriate context.

File: lib/common/buffer-utils.ts [hunk 6]
```diff
@@ -38,19 +35,25 @@ function getBuffer (value) {
...
+  data: Buffer;
+  length: number | undefined;
+}
+
+export function bufferToMeta (value: BufferLike): BufferMeta {
   return {
     type: getType(value),
     data: getBuffer(value),
-    length: value.length
+    length: (value as Buffer).length
```
Comment (nornagon) [inline lib/common/buffer-utils.ts]:
```suggestion
    // NB. We only use length when decoding Int8Array and friends.
    // For other buffer-like types this is expected to be undefined.
    length: (value as Buffer).length
```

File: lib/common/buffer-utils.ts (hunk 2 above)
Comment (nornagon) [reply lib/common/buffer-utils.ts]: nvm i went ahead and added it :)

**Pull request merged**